sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -310,3 +312,28 @@ case class StringLength(child: Expression) extends UnaryExpression with ExpectsI
     defineCodeGen(ctx, ev, c => s"($c).length()")
   }
 }
+
+/**
+ *  A function that return soundex name of a ID
+ */
+case class SoundEx(child: Expression) extends UnaryExpression with ExpectsInputTypes{
+  val so = new Soundex
+  override def dataType: DataType = StringType
+  override def expectedChildTypes: Seq[DataType] = Seq(StringType)
+  override def eval(input: InternalRow): Any = {
+    val string  = child.eval(input)
+    if (string == null) null
+    else {
+      val str: java.lang.String = string.toString
+      so.encode(str)
```

Comment:
The return value should be in `UTF8String` for `StringType`.
`UTF8String.fromString(so.encode(str))`